src/decode.js

```diff
@@ -100,5 +100,30 @@ function decode (options) {
   })()
 }
 
+function decodeFromReader (reader, options) {
+  options = options || {}
+
+  let byteLength = 1 // Read single byte chunks until the length is known
+  const varByteSource = {
+    [Symbol.asyncIterator] () { return this },
+    next: () => {
+      return reader.next(byteLength)
+        .catch(() => {
+          // There was no data left to read, end the iterator
+          return { done: true, value: null }
```

Comment:
Can we assert that is this an [under read](https://github.com/alanshaw/it-reader/blob/master/index.js#L38) error and not smother all other errors?

Reply:
We're checking the error code now and throwing if it's not an under read.

Perhaps there's a change we could make in it-reader to resolve this. What's happening is that we are technically done reading the data, but decodeFromReader will always try to read the next byte since `it-reader` returned `done:false` on the previous `next` call. If `it-reader` said, here are the 24 bytes you requested, and also I don't have anymore, we could avoid that check. I need to look at the it-reader code to see if that's doable.

Reply:
If this is to be used for reading multiple messages then it's fine like this.